### PR TITLE
feat: dropdown multiselect component

### PR DESCRIPTION
**Pull request by hadalin**
Duplicate of https://github.com/reciprocity/zenlib/pull/51. This PR is not from a forked repo ([explanation](https://github.com/reciprocity/zenlib/pull/51#issuecomment-573001656)).

### CODE REVIEW

**Comment by zenbot-0** [comment]
🔬 [Click here](https://reciprocity.github.io/zenlib/branches/dropdown-multiselect) to test your changes on Storybook

File: packages/dropdown-multiselect/DropdownMultiselect.vue [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  background: $white;
+  color: $black;
+}
+.multiselect__option--highlight:after {
+  display: none;
+}
+.multiselect__option--selected {
+  font-weight: normal;
+}
+.multiselect__option--selected.multiselect__option--highlight {
```
Comment (leferreyra) [inline packages/dropdown-multiselect/DropdownMultiselect.vue]: About the styling thing. I think you could try to wrap your component in another class, so specificity increases so the other styles don't override yours. Ideally, you'd do that for the other multi-select component too.

File: packages/_storybook_/stories/DropdownMultiselect.stories.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { storiesOf } from "@storybook/vue";
+import { action } from "@storybook/addon-actions";
+import DropdownMultiselect from "@reciprocity/dropdown-multiselect";
+import README from "../../dropdown-multiselect/README.md";
+
+storiesOf("DropdownMultiselect", module)
```
Comment (leferreyra) [inline packages/_storybook_/stories/DropdownMultiselect.stories.js]: Have you considered updating the current `multiselect` component with these new features instead of adding a new component?

It looks like all these new features are only for presentation, no actual behavior changes.

I'm not saying this is incorrect just wanna know if you considered that an option since they both use the same `vue-multiselect` underlying component.

Comment (hadalin) [reply packages/_storybook_/stories/DropdownMultiselect.stories.js]: >just wanna know if you considered that an option since they both use the same vue-multiselect underlying component.

Yes, we considered that option, but I, @pr1moz and @mplesec agreed to create a new component, the reason being it would complicate things too much if we tried adjusting Multiselect to look the same as dropdown from the design specs.

Comment (leferreyra) [reply packages/_storybook_/stories/DropdownMultiselect.stories.js]: Ok, fair enough.

**Comment by pr1moz** [comment]
@hadalin https://github.com/reciprocity/zenlib/pull/51#pullrequestreview-341124208

**Comment by pr1moz** [comment]
Posting here as well.

> hadalin we discussed this with mplesec on Friday - I think it's better to implement the current look for now and change it as part of the redesign effort. We want to avoid inconsistencies until we have dedicated time from product to systematically change the look of related components (forms in this case).
This is something that needs to be discussed with Klara today as well.

File: packages/multiselect/Multiselect.vue [hunk 3]
```diff
@@ -1,6 +1,7 @@
 <template>
   <vue-multiselect
     :name="name"
+    class="zen-multiselect"
```
Comment (leferreyra) [inline packages/multiselect/Multiselect.vue]: I think these `Multiselect` component changes should be in a separate PR.

File: packages/multiselect/Multiselect.vue [hunk 4]
```diff
@@ -207,516 +208,519 @@ $vue-ms-spinner-color: $dark;
 
 $title-truncate-width: 50ch;
 
-// --------------------------------
-// Select field
-// --
+.zen-multiselect {
```
Comment (leferreyra) [inline packages/multiselect/Multiselect.vue]: For future reference, please avoid the use of the word zen within the library.

Comment (leferreyra) [reply packages/multiselect/Multiselect.vue]: May I suggest something along the lines of `.multiselect-wrapper` or `.multiselect-container`

Comment (hadalin) [reply packages/multiselect/Multiselect.vue]: Vue loader has a thing called [Scoped CSS](https://vue-loader.vuejs.org/guide/scoped-css.html) which prevents styles from leaking. Maybe we should investigate if that's possible in zenlib.

Comment (hadalin) [reply packages/multiselect/Multiselect.vue]: Also, should we refactor [this](https://github.com/reciprocity/zenlib/blob/ca7567429e6a66dd75ab02b32f53661e5ecbb73a/packages/multiselect/Multiselect.vue#L78) Zen occurrence too?

Comment (leferreyra) [reply packages/multiselect/Multiselect.vue]: Yeah, there's an option in the rollup vue plugin for scoped css, would be great to implement that, I'd say PRs welcome ;)

https://rollup-plugin-vue.vuejs.org/options.html#style-postcssmodulesoptions

Comment (leferreyra) [reply packages/multiselect/Multiselect.vue]: Sure, if you can and want to remove current occurrences of zen would be awesome. I'm happy with not adding more new stuff with zen though.

Comment (hadalin) [reply packages/multiselect/Multiselect.vue]: In the docs it says

>And many other features, maintaining parity with Vue Loader.

so I guess scoping is enabled by default and you enable it with `<style lang="scss" scoped>`, but it doesn't seem to be working.

Comment (hadalin) [reply packages/multiselect/Multiselect.vue]: Wait, no, `<style lang="scss" scoped>` is working. The `dist.js` is correctly updated

Screenshot 2020-01-13 at 15 34 02

Comment (hadalin) [reply packages/multiselect/Multiselect.vue]: Unrelated to this PR, but anyway, I've enabled `scoped>` for `Multiselect` and the component doesn't render properly. I'm guessing there's something wrong with rules in `<style lang="scss" scoped>`. I'm thinking `.multiselect` class in there is causing problems.

Comment (hadalin) [reply packages/multiselect/Multiselect.vue]: It would be nice to resolve this issue. That way we wouldn't have to use `.multiselect-wrapper` which is an unneeded overhead.

Comment (leferreyra) [reply packages/multiselect/Multiselect.vue]: Agree 100%. I created a ticket for this https://reciprocitylabs.atlassian.net/browse/ZEN-12877
